src/Aspire.Cli/Backchannel/AuxiliaryBackchannelMonitor.cs

```diff
@@ -35,7 +35,24 @@ internal sealed class AuxiliaryBackchannelMonitor(
     /// <summary>
     /// Gets or sets the path to the selected AppHost. When set, this AppHost will be used for MCP operations.
     /// </summary>
-    public string? SelectedAppHostPath { get; set; }
+    public string? SelectedAppHostPath
+    {
+        get => _selectedAppHostPath;
+        set
+        {
+            if (_selectedAppHostPath != value)
+            {
+                _selectedAppHostPath = value;
+                SelectedAppHostChanged?.Invoke();
+            }
+        }
+    }
+    private string? _selectedAppHostPath;
+
+    /// <summary>
+    /// Event raised when the selected AppHost changes.
+    /// </summary>
+    public event Action? SelectedAppHostChanged;
```

Comment:
I think this should be changed to have SelectedAppHostPath with a getter only, and an async method for set. Then inside the set method it executes handlers that are registered for the app host changing. That eliminates running async code in the event handler by calling `Task.Run`.

Registering a handler can be like cancellation token. i.e. register a delegate and get back an IDisposable. Dispose to remove handler.